datasimulator/graph.py

```diff
@@ -203,53 +203,36 @@ def construct_graph_edges(self):
 
     def generate_submission_order_path_to_node(self, node, cmc_node=None):
         """
-        Generate submission order so that the current node can be submitted
-
-        Args:
-            node(Node): current node object
-
-        Outputs:
-            list: list of submission order
-
+        From the specified `end_node`, step through the graph from bottom to top to generate the minimal
+        submission order to that node.
         """
-        submission_order = [node]
-        if cmc_node:
-            submission_order.append(cmc_node)
-        index = 0
-
-        project_node = None
-        while index < len(submission_order):
-            cur_node = submission_order[index]
-            index += 1
-            if not cur_node:
-                continue
-            for linked_node_dict in cur_node.required_links:
-                if linked_node_dict["node"].name == "project":
-                    project_node = linked_node_dict["node"]
-                    continue
-                if linked_node_dict["node"] in submission_order:
-                    # reorder to place the node at the beginning. eg. if this node is parent to 2 other
-                    # nodes, we need it to be submitted before the 2 others, and not in-between.
-                    submission_order.remove(linked_node_dict["node"])
-                submission_order.append(linked_node_dict["node"])
-
-        if project_node:  # project should always be submitted first
-            submission_order.append(project_node)
-
+        # get the bottom-to-top path from the node, and then reverse since the actual submission order
+        # must be top-to-bottom (parent nodes first)
+        submission_order = get_graph_traversal_path(direction="up", start_node=node)
         submission_order.reverse()
 
+        # if specified, make sure that `core_metadata_collection` is in the submission order
+        if cmc_node and cmc_node not in submission_order:
```

Comment:
what is the end doing in this case?

Reply:
it's the same logic as the current code: https://github.com/uc-cdis/data-simulator/blob/288b5f2/datasimulator/graph.py#L216-L217

we can provide the `cmc_node` parameter to force the function to add `core_metadata_collection` to the  submission order. that's because the gen3-qa tests expect it to be present